code/datums/components/crafting/crafting.dm

```diff
@@ -172,7 +172,9 @@
 	var/send_feedback = 1
 	if(check_contents(a, R, contents))
 		if(check_tools(a, R, contents))
-			//If we're a mob we'll try a do_after; non mobs will instead instantly construct the item
+			//If we're a mob and they have the construction trait we'll try a do_after with a construction benefit; non mobs will instead instantly construct the item
+			if(ismob(a) && HAS_TRAIT(a, TRAIT_CONSTRUCTION) && !do_after(a, R.time/2, target = a))
+				return "."
 			if(ismob(a) && !do_after(a, R.time, target = a))
 				return "."
```

Comment:
Ironically TRAIT_CONSTRUCTION will slow you down here since the return doesn't complete crafting, it stops crafting so this will run for time/2 seconds, finish and then run for time seconds

Reply:
Resolved by 
c749b4c